control/tests/xferfcn_test.py

```diff
@@ -467,21 +466,36 @@ def test_evalfr_mimo(self):
                 [-0.083333333333333, -0.188235294117647 - 0.847058823529412j,
                  -1. - 8.j]]
 
-        np.testing.assert_array_almost_equal(sys._evalfr(2.), resp)
+        np.testing.assert_array_almost_equal(evalfr(sys, 2j), resp)
 
         # Test call version as well
         np.testing.assert_array_almost_equal(sys(2.j), resp)
 
-    def test_freqresp_siso(self):
-        """Evaluate the SISO magnitude and phase at multiple frequencies"""
+    def test_freqresp_deprecated(self):
+        sys = TransferFunction([1., 3., 5], [1., 6., 2., -1.])
+        # Deprecated version of the call (should generate warning)
+        import warnings
+        with warnings.catch_warnings(record=True) as w:
+            # Set up warnings filter to only show warnings in control module
+            warnings.filterwarnings("ignore")
+            warnings.filterwarnings("always", module="control")
+
+            # Make sure that we get a pending deprecation warning
+            sys.freqresp(1.)
+            assert issubclass(w[-1].category, DeprecationWarning)
```

Comment:
use `pytest.warns`

Reply:
willhave to leave to richard to change actuak code

Reply:
3b10af001bf07644062c0c2da0067bc501cd3eff